lib/SPIRV/SPIRVReader.cpp

```diff
@@ -487,8 +487,10 @@ std::string SPIRVToLLVM::transTypeToOCLTypeName(SPIRVType *T, bool IsSigned) {
     case 64:
       return Prefix + "long";
     default:
+      // Arbitrary precision integer
+      if (T->getIntegerBitWidth() < 64)
```

Comment:
But SPIR-V translator currently limits int size by 64 bits (see class SPIRVTypeInt validate method).

Reply:
Hmm, but it looks like this is the only place which limits int size. In release mode translation works good with bigger length integers. Should I remove upper limit check in SPIRVTypeInt?

Reply:
From my perspective these limits shall be applied only during forward translation, and not applicable on reversed translation, since it shall accept whatever SPIR-V module, and it's not a reversed translator's duty to check module's correctness.

Reply:
Check was removed but also due to the fact that SPIRVTypeInt validates bitwidth in reverse translation too before the call of transTypeToOCLTypeName.